python_modules/dagster/dagster/core/definitions/sensor_definition.py

```diff
@@ -338,13 +340,13 @@ def evaluate_tick(self, context: "SensorEvaluationContext") -> "SensorExecutionD
                     check.failed("Expected a single SkipReason: received multiple SkipReasons")
 
             if has_run_request:
-                run_requests = result
+                run_requests = cast(List[RunRequest], result)
```

Comment:
@prha - I'm not super familiar with this code, but I wonder if it could benefit from me adding a check that `has_run_request` and `has_run_reaction` are not both true?

Reply:
@yuhan